### ISSUE TITLE
Chart title property should be dynamic

### ISSUE THREAD

**Issue by AlexandreSajus** [Avaiga/taipy#2371]
### What went wrong? 🤔

Updating dynamically the title property of a chart visual element does not work. Here is an example where I try to change the title of a chart using a button. Pressing the button should change the title but does nothing.

```python
import pandas as pd

import taipy.gui.builder as tgb
from taipy.gui import Gui

data = pd.DataFrame({"Product": ["Shovel", "Rake", "Hoe"], "Price": [10, 5, 7]})

title = "Undefined"


def update_title(state):
    state.title = "Price of Gardening Tools"


with tgb.Page() as page:
    tgb.button("Update Title", on_action=update_title)
    tgb.chart(data="{data}", title="{title}")

Gui(page).run()
```

Using layout is a workaround:
```python
import pandas as pd

import taipy.gui.builder as tgb
from taipy.gui import Gui

data = pd.DataFrame({"Product": ["Shovel", "Rake", "Hoe"], "Price": [10, 5, 7]})

layout = {"title": "Undefined"}


def update_title(state):
    state.layout = {"title": "Price of Gardening Tools"}


with tgb.Page() as page:
    tgb.button("Update Title", on_action=update_title)
    tgb.chart(data="{data}", layout="{layout}")

Gui(page).run()
```

![image](https://github.com/user-attachments/assets/4ea21037-5e6e-4a24-a6b1-33a390444a27)

### Runtime Environment

Windows 11

### Browsers

Chrome

### OS

Windows

### Version of Taipy

4.0.2

### Acceptance Criteria

- [ ] A unit test reproducing the bug is added.
- [ ] Any new code is covered by a unit tested.
- [ ] Check code coverage is at least 90%.
- [ ] The bug reporter validated the fix.
- [ ] Related issue(s) in taipy-doc are created for documentation and Release Notes are updated.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by jrobinAV** [comment]
I believe this is not a malfunction but an improvement as the property is probably not dynamic.

**Comment by ChamaliVishmani** [comment]
Hi is anyone working on this issue? If not, can I start working on this?

**Comment by FlorianJacta** [comment]
Yes, sure! If you need any help, please ask to our R&D @FabienLelaquais and @FredLL-Avaiga !

**Comment by ChamaliVishmani** [comment]
Okay, thanks !

**Comment by ADORA-ATAYA** [comment]
Hi is anyone working on this issue? If not, can I start working on this?

**Comment by FlorianJacta** [comment]
Maybe, @ChamaliVishmani is already traying to solve the issue but you can be assigned as well @ADORA-ATAYA

**Comment by ADORA-ATAYA** [comment]
thanks

**Comment by ChamaliVishmani** [comment]
Hi @FlorianJacta, I'm still working on this issue. I attempted to use useState to handle the title prop, as the issue seems to be related to its immutability. I also explored other approaches, including modifying the prop directly, but haven't found a working solution yet.

Could you provide some guidance on debugging and potential ways to fix this? Any suggestions would be greatly appreciated!

**Comment by ChamaliVishmani** [comment]
Hi, @ADORA-ATAYA I’d be happy to collaborate if you’d like to help.

**Comment by ADORA-ATAYA** [comment]
Sure @ChamaliVishmani, I am happy to collaborate with you

**Comment by FlorianJacta** [comment]
@FabienLelaquais or @FredLL-Avaiga, could you provide some guidance to @ChamaliVishmani for this issue?

**Comment by FredLL-Avaiga** [comment]
> [@FabienLelaquais](https://github.com/FabienLelaquais) or [@FredLL-Avaiga](https://github.com/FredLL-Avaiga), could you provide some guidance to [@ChamaliVishmani](https://github.com/ChamaliVishmani) for this issue?

The property type should be changed to `PropertyType.dynamic_string` in [factory.py](https://github.com/Avaiga/taipy/blob/7af14bdcb51912e9d54b128e18526e274a4473cb/taipy/gui/_renderers/factory.py#L131)

Then on the front-end side, the attribute has to be managed as other Dynamic string properties.
